djcelery/management/base.py

```diff
@@ -35,15 +35,20 @@ def _init(self, *args, **kwargs):
                 __old__init__(self, *args, **kwargs)
                 self._thread_ident = _get_ident()
 
+            def _validate_thread(self):
+                is_valid = self.allow_thread_sharing
+                is_valid = is_valid or self._thread_ident == _get_ident()
+                return is_valid
+
             def _validate_thread_sharing(self):
-                if (not self.allow_thread_sharing and
-                        self._thread_ident != _get_ident()):
+                if not self.validate_thread():
                     raise DatabaseError(
                         DB_SHARED_THREAD % (
                             self.alias, self._thread_ident, _get_ident()),
                     )
 
             BaseDatabaseWrapper.__init__ = _init
+            BaseDatabaseWrapper.validate_thread = _validate_thread
```

Comment:
@liquidpele , it doesn't have underscore here.

Reply:
Oh, I see.  Didn't expand to look at the whole file.  Thx.